internal/server/boltdbstate/job.go

```diff
@@ -1321,6 +1321,11 @@ func (s *State) jobCreate(dbTxn *bolt.Tx, memTxn *memdb.Txn, jobpb *pb.Job) erro
 
 		// Go through and ensure that each exists.
 		for _, id := range jobpb.DependsOn {
+			if id == "" {
+				return status.Errorf(codes.FailedPrecondition,
+					"job %q has an empty string id for depends_on", jobpb.Id)
```

Comment:
Good catch! I'm wondering how this could happen... I can't recall when I've seen a job without an ID 🤔

Reply:
This came about from a mistake in my own code! I was adding empty string to the `DependsOn` slice in the job template generator (this was a bug) and figured we could use some extra validation here that this shouldn't be allowed.